pkg/batchfetcher/batchfetcher.go

```diff
@@ -104,16 +106,12 @@ func NewModule(mc *ModuleConfig, epochNr t.EpochNr, clientProgress *clientprogre
 
 	// The AppRestoreState handler restores the batch fetcher's state from a checkpoint
 	// and forwards the event to the application, so it can restore its state too.
-	eventpbdsl.UponAppRestoreState(m, func(restoreState *eventpb.AppRestoreState) error {
-
-		// Load the checkpoint from the received event.
-		chkp := checkpoint.StableCheckpointFromPb(restoreState.Checkpoint)
-
+	eventpbdsl.UponAppRestoreState(m, func(chkp *checkpointpb.StableCheckpoint) error {
 		// Update current epoch number.
-		epochNr = chkp.Epoch()
+		epochNr = t.EpochNr(chkp.Snapshot.EpochData.EpochConfig.EpochNr)
```

Comment:
If we keep using the checkpoint abstraction, we don't need to rely on the internal format of the data structure (in case it ever changes). And the code will also be easier to look at.

Reply:
ditto, need to come back to this to tag the type on protobufs and thus removing `t.EpochNr` once we are back to using the codegen code.

Reply:
Update: after addressing @xosmig 's comments I decided to also add support for map types defining only the key or the value type. So in a new upcoming PR we can clean up this code further. Not doing it on this PR for the purpose of not overcomplicating it at this stage.